[BUG] - Android TV watch next crash

### Description

Getting a crash upon selecting a episode from the watch next Android TV screen. Happened to me on 2 seperate occasions with the same item. First try it crashed and then I decided to watch something else. The next day tried again and it crashed again. Starting the episode from Wholphin works without issue.

### Reproduction steps

Select episode from watch next on Android TV home screen > crash. If I directly try it again after the crash it works without issue.

### App Version

v0.3.12-1

### Server Version

10.11.2

### Device

NVIDIA SHIELD TV PRO 2019

### Logs

```bash
---
Date: Tue Jan 20 15:20:58 GMT+01:00 2026
---
{
  "STACK_TRACE": "java.lang.RuntimeException: Unable to resume activity {com.github.damontecres.wholphin\/com.github.damontecres.wholphin.MainActivity}: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0\n\tat android.app.ActivityThread.performResumeActivity(ActivityThread.java:3835)\n\tat android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3867)\n\tat android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:51)\n\tat android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145)\n\tat android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)\n\tat android.app.ActivityThread$H.handleMessage(ActivityThread.java:1809)\n\tat android.os.Handler.dispatchMessage(Handler.java:106)\n\tat android.os.Looper.loop(Looper.java:193)\n\tat android.app.ActivityThread.main(ActivityThread.java:6721)\n\tat java.lang.reflect.Method.invoke(Native Method)\n\tat com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)\n\tat com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)\nCaused by: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0\n\tat java.util.ArrayList.set(ArrayList.java:453)\n\tat com.github.damontecres.wholphin.services.NavigationManager.replace(NavigationManager.kt:78)\n\tat com.github.damontecres.wholphin.MainActivity.onNewIntent(MainActivity.kt:341)\n\tat android.app.Activity.performNewIntent(Activity.java:7192)\n\tat android.app.Instrumentation.callActivityOnNewIntent(Instrumentation.java:1366)\n\tat android.app.Instrumentation.callActivityOnNewIntent(Instrumentation.java:1378)\n\tat android.app.ActivityThread.deliverNewIntents(ActivityThread.java:3150)\n\tat android.app.ActivityThread.performResumeActivity(ActivityThread.java:3820)\n\t... 11 more\n",
  "LOGCAT": "01-20 14:46:37.520 D\/WM-SystemJobService( 5720): onStopJob for WorkGenerationalId(workSpecId=06661093-d98a-41b0-b658-1fb0087a3a2b, generation=0)\n01-20 14:46:37.520 D\/WM-GreedyScheduler( 5720): Starting work for 06661093-d98a-41b0-b658-1fb0087a3a2b\n01-20 14:46:37.526 D\/WM-SystemJobService( 5720): onStartJob for WorkGenerationalId(workSpecId=06661093-d98a-41b0-b658-1fb0087a3a2b, generation=0)\n01-20 14:46:37.527 D\/WM-Processor( 5720): Processor: processing WorkGenerationalId(workSpecId=06661093-d98a-41b0-b658-1fb0087a3a2b, generation=0)\n01-20 14:46:37.532 D\/WM-Processor( 5720): WorkerWrapper interrupted for 06661093-d98a-41b0-b658-1fb0087a3a2b\n01-20 14:46:37.532 D\/WM-StopWorkRunnable( 5720): StopWorkRunnable for 06661093-d98a-41b0-b658-1fb0087a3a2b; Processor.stopWork = true\n01-20 14:46:37.534 D\/WM-Processor( 5720): Processor: processing WorkGenerationalId(workSpecId=06661093-d98a-41b0-b658-1fb0087a3a2b, generation=0)\n01-20 14:46:37.536 D\/WM-Processor( 5720): Work WorkGenerationalId(workSpecId=06661093-d98a-41b0-b658-1fb0087a3a2b, generation=0) is already enqueued for processing\n01-20 14:46:37.536 D\/WM-WorkerWrapper( 5720): Status for 06661093-d98a-41b0-b658-1fb0087a3a2b is ENQUEUED; not doing any work and rescheduling for later execution\n01-20 14:46:37.538 D\/WM-Processor( 5720): Processor 06661093-d98a-41b0-b658-1fb0087a3a2b executed; reschedule = true\n01-20 14:46:37.538 D\/WM-SystemJobService( 5720): 06661093-d98a-41b0-b658-1fb0087a3a2b executed on JobScheduler\n01-20 14:46:37.550 D\/NetworkSecurityConfig( 5720): Using Network Security Config from resource network_security_config debugBuild: false\n01-20 14:46:37.572 D\/WM-WorkerWrapper( 5720): Starting work for com.github.damontecres.wholphin.services.tvprovider.TvProviderWorker\n01-20 14:46:37.578 D\/WM-GreedyScheduler( 5720): Cancelling work ID 06661093-d98a-41b0-b658-1fb0087a3a2b\n01-20 14:46:37.583 D\/WM-PackageManagerHelper( 5720): Skipping component enablement for androidx.work.impl.background.systemalarm.RescheduleReceiver\n01-20 14:46:37.585 I\/OkHttpClient( 5720): GET http:\/\/192.168.1.102:8096\/Users\/Me\n01-20 14:46:37.706 I\/OkHttpClient( 5720): GET http:\/\/192.168.1.102:8096\/System\/Info\/Public\n01-20 14:46:37.722 I\/OkHttpClient( 5720): GET http:\/\/192.168.1.102:8096\/UserItems\/Resume?userId=a503825e-bd30-4ec7-b238-fe95566186cf&limit=10&fields=PrimaryImageAspectRatio&fields=SeasonUserData&fields=ChildCount&fields=Overview&fields=SortName&fields=MediaSourceCount&includeItemTypes=Movie&includeItemTypes=Episode&includeItemTypes=Series&includeItemTypes=Video&includeItemTypes=Season&includeItemTypes=CollectionFolder&includeItemTypes=Folder&includeItemTypes=UserView&includeItemTypes=Trailer&includeItemTypes=TvChannel&includeItemTypes=TvProgram&includeItemTypes=LiveTvChannel&includeItemTypes=LiveTvProgram&includeItemTypes=Recording&enableTotalRecordCount=true&enableImages=true&excludeActiveSessions=false\n01-20 14:46:37.743 I\/OkHttpClient( 5720): GET http:\/\/192.168.1.102:8096\/Shows\/NextUp?userId=a503825e-bd30-4ec7-b238-fe95566186cf&limit=10&fields=PrimaryImageAspectRatio&fields=SeasonUserData&fields=ChildCount&fields=Overview&fields=SortName&fields=MediaSourceCount&imageTypeLimit=1&enableUserData=true&enableTotalRecordCount=true&disableFirstEpisode=false&enableResumable=false&enableRewatching=false\n01-20 14:46:37.828 I\/OkHttpClient( 5720): GET http:\/\/192.168.1.102:8096\/Items\/Latest?fields=PrimaryImageAspectRatio&fields=SeasonUserData&fields=ChildCount&fields=Overview&fields=SortName&fields=MediaSourceCount&imageTypeLimit=1&limit=25&groupItems=true\n01-20 14:46:37.926 I\/WM-WorkerWrapper( 5720): Worker result SUCCESS for Work [ id=06661093-d98a-41b0-b658-1fb0087a3a2b, tags={ com.github.damontecres.wholphin.services.tvprovider.TvProviderWorker } ]\n01-20 14:46:37.929 D\/WM-Processor( 5720): Processor 06661093-d98a-41b0-b658-1fb0087a3a2b executed; reschedule = false\n01-20 14:46:37.930 D\/WM-GreedyScheduler( 5720): Cancelling work ID 06661093-d98a-41b0-b658-1fb0087a3a2b\n01-20 14:46:37.934 D\/WM-SystemJobScheduler( 5720): Scheduling work ID 06661093-d98a-41b0-b658-1fb0087a3a2bJob ID 310\n01-20 14:46:42.423 I\/tecres.wholphi( 5720): Waiting for a blocking GC ProfileSaver\n01-20 14:46:42.431 I\/tecres.wholphi( 5720): Waiting for a blocking GC ProfileSaver\n01-20 14:46:42.432 I\/tecres.wholphi( 5720): WaitForGcToComplete blocked ProfileSaver on HeapTrim for 8.654ms\n01-20 14:46:42.441 I\/tecres.wholphi( 5720): WaitForGcToComplete blocked HeapTrim on ProfileSaver for 9.024ms\n01-20 15:20:52.996 E\/tecres.wholphi(10912): Not starting debugger since process cannot load the jdwp agent.\n01-20 15:20:53.071 W\/tecres.wholphi(10912): JIT profile information will not be recorded: profile file does not exits.\n01-20 15:20:53.071 I\/chatty  (10912): uid=10081(com.github.damontecres.wholphin) identical 3 lines\n01-20 15:20:53.071 W\/tecres.wholphi(10912): JIT profile information will not be recorded: profile file does not exits.\n01-20 15:20:53.104 I\/ACRA    (10912): ACRA is enabled for com.github.damontecres.wholphin, initializing...\n01-20 15:20:53.154 D\/AppCompatDelegate(10912): Checking for metadata for AppLocalesMetadataHolderService : Service not found\n01-20 15:20:53.294 D\/NetworkSecurityConfig(10912): Using Network Security Config from resource network_security_config debugBuild: false\n01-20 15:20:53.307 D\/WM-PackageManagerHelper(10912): Skipping component enablement for androidx.work.impl.background.systemjob.SystemJobService\n01-20 15:20:53.307 D\/WM-Schedulers(10912): Created SystemJobScheduler and enabled SystemJobService\n01-20 15:20:53.348 D\/WM-ForceStopRunnable(10912): The default process name was not specified.\n01-20 15:20:53.349 D\/WM-ForceStopRunnable(10912): Performing cleanup operations.\n01-20 15:20:53.351 I\/Wholphin(10912): MainActivity.onCreate: savedInstanceState is null=false\n01-20 15:20:53.366 D\/WM-SystemJobScheduler(10912): Reconciling jobs\n01-20 15:20:53.373 D\/WM-ForceStopRunnable(10912): Found unfinished work, scheduling it.\n01-20 15:20:53.380 W\/tecres.wholphi(10912): Accessing hidden method Landroid\/media\/MediaFormat;->getInteger(Ljava\/lang\/String;I)I (dark greylist, linking)\n01-20 15:20:53.394 W\/AudioCapabilities(10912): Unsupported mime audio\/mpeg-L2\n01-20 15:20:53.398 I\/OkHttpClient(10912): GET http:\/\/192.168.1.102:8096\/Users\/Me\n01-20 15:20:53.403 W\/AudioCapabilities(10912): Unsupported mime audio\/x-ms-wma\n01-20 15:20:53.403 W\/tecres.wholphi(10912): Accessing hidden method Landroid\/view\/View;->setTransitionVisibility(I)V (dark greylist, linking)\n01-20 15:20:53.406 D\/AndroidRuntime(10912): Shutting down VM\n01-20 15:20:53.407 E\/AndroidRuntime(10912): FATAL EXCEPTION: main\n01-20 15:20:53.407 E\/AndroidRuntime(10912): Process: com.github.damontecres.wholphin, PID: 10912\n01-20 15:20:53.407 E\/AndroidRuntime(10912): java.lang.RuntimeException: Unable to resume activity {com.github.damontecres.wholphin\/com.github.damontecres.wholphin.MainActivity}: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0\n01-20 15:20:53.407 E\/AndroidRuntime(10912): \tat android.app.ActivityThread.performResumeActivity(ActivityThread.java:3835)\n01-20 15:20:53.407 E\/AndroidRuntime(10912): \tat android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3867)\n01-20 15:20:53.407 E\/AndroidRuntime(10912): \tat android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:51)\n01-20 15:20:53.407 E\/AndroidRuntime(10912): \tat android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145)\n01-20 15:20:53.407 E\/AndroidRuntime(10912): \tat android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)\n01-20 15:20:53.407 E\/AndroidRuntime(10912): \tat android.app.ActivityThread$H.handleMessage(ActivityThread.java:1809)\n01-20 15:20:53.407 E\/AndroidRuntime(10912): \tat android.os.Handler.dispatchMessage(Handler.java:106)\n01-20 15:20:53.407 E\/AndroidRuntime(10912): \tat android.os.Looper.loop(Looper.java:193)\n01-20 15:20:53.407 E\/AndroidRuntime(10912): \tat android.app.ActivityThread.main(ActivityThread.java:6721)\n01-20 15:20:53.407 E\/AndroidRuntime(10912): \tat java.lang.reflect.Method.invoke(Native Method)\n01-20 15:20:53.407 E\/AndroidRuntime(10912): \tat com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)\n01-20 15:20:53.407 E\/AndroidRuntime(10912): \tat com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)\n01-20 15:20:53.407 E\/AndroidRuntime(10912): Caused by: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0\n01-20 15:20:53.407 E\/AndroidRuntime(10912): \tat java.util.ArrayList.set(ArrayList.java:453)\n01-20 15:20:53.407 E\/AndroidRuntime(10912): \tat com.github.damontecres.wholphin.services.NavigationManager.replace(NavigationManager.kt:78)\n01-20 15:20:53.407 E\/AndroidRuntime(10912): \tat com.github.damontecres.wholphin.MainActivity.onNewIntent(MainActivity.kt:341)\n01-20 15:20:53.407 E\/AndroidRuntime(10912): \tat android.app.Activity.performNewIntent(Activity.java:7192)\n01-20 15:20:53.407 E\/AndroidRuntime(10912): \tat android.app.Instrumentation.callActivityOnNewIntent(Instrumentation.java:1366)\n01-20 15:20:53.407 E\/AndroidRuntime(10912): \tat android.app.Instrumentation.callActivityOnNewIntent(Instrumentation.java:1378)\n01-20 15:20:53.407 E\/AndroidRuntime(10912): \tat android.app.ActivityThread.deliverNewIntents(ActivityThread.java:3150)\n01-20 15:20:53.407 E\/AndroidRuntime(10912): \tat android.app.ActivityThread.performResumeActivity(ActivityThread.java:3820)\n01-20 15:20:53.407 E\/AndroidRuntime(10912): \t... 11 more\n01-20 15:20:53.407 E\/ACRA    (10912): ACRA caught a RuntimeException for com.github.damontecres.wholphin\n01-20 15:20:53.407 E\/ACRA    (10912): java.lang.RuntimeException: Unable to resume activity {com.github.damontecres.wholphin\/com.github.damontecres.wholphin.MainActivity}: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0\n01-20 15:20:53.407 E\/ACRA    (10912): \tat android.app.ActivityThread.performResumeActivity(ActivityThread.java:3835)\n01-20 15:20:53.407 E\/ACRA    (10912): \tat android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3867)\n01-20 15:20:53.407 E\/ACRA    (10912): \tat android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:51)\n01-20 15:20:53.407 E\/ACRA    (10912): \tat android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145)\n01-20 15:20:53.407 E\/ACRA    (10912): \tat android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)\n01-20 15:20:53.407 E\/ACRA    (10912): \tat android.app.ActivityThread$H.handleMessage(ActivityThread.java:1809)\n01-20 15:20:53.407 E\/ACRA    (10912): \tat android.os.Handler.dispatchMessage(Handler.java:106)\n01-20 15:20:53.407 E\/ACRA    (10912): \tat android.os.Looper.loop(Looper.java:193)\n01-20 15:20:53.407 E\/ACRA    (10912): \tat android.app.ActivityThread.main(ActivityThread.java:6721)\n01-20 15:20:53.407 E\/ACRA    (10912): \tat java.lang.reflect.Method.invoke(Native Method)\n01-20 15:20:53.407 E\/ACRA    (10912): \tat com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)\n01-20 15:20:53.407 E\/ACRA    (10912): \tat com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)\n01-20 15:20:53.407 E\/ACRA    (10912): Caused by: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0\n01-20 15:20:53.407 E\/ACRA    (10912): \tat java.util.ArrayList.set(ArrayList.java:453)\n01-20 15:20:53.407 E\/ACRA    (10912): \tat com.github.damontecres.wholphin.services.NavigationManager.replace(NavigationManager.kt:78)\n01-20 15:20:53.407 E\/ACRA    (10912): \tat com.github.damontecres.wholphin.MainActivity.onNewIntent(MainActivity.kt:341)\n01-20 15:20:53.407 E\/ACRA    (10912): \tat android.app.Activity.performNewIntent(Activity.java:7192)\n01-20 15:20:53.407 E\/ACRA    (10912): \tat android.app.Instrumentation.callActivityOnNewIntent(Instrumentation.java:1366)\n01-20 15:20:53.407 E\/ACRA    (10912): \tat android.app.Instrumentation.callActivityOnNewIntent(Instrumentation.java:1378)\n01-20 15:20:53.407 E\/ACRA    (10912): \tat android.app.ActivityThread.deliverNewIntents(ActivityThread.java:3150)\n01-20 15:20:53.407 E\/ACRA    (10912): \tat android.app.ActivityThread.performResumeActivity(ActivityThread.java:3820)\n01-20 15:20:53.407 E\/ACRA    (10912): \t... 11 more\n01-20 15:20:53.420 D\/WM-SystemJobScheduler(10912): Scheduling work ID 06661093-d98a-41b0-b658-1fb0087a3a2bJob ID 310\n01-20 15:20:53.426 W\/VideoCapabilities(10912): Unsupported mime image\/vnd.android.heic\n",
  "CUSTOM_DATA": {
    "SDK_INT": "28"
  },
  "IS_SILENT": false,
  "REPORT_ID": "003307ee-46fd-48cb-a32e-58e8bde5dd46",
  "PHONE_MODEL": "SHIELD Android TV",
  "ANDROID_VERSION": "9",
  "BRAND": "NVIDIA",
  "SHARED_PREFERENCES": {
    "default": {
      "version.current.name": "0.3.12-1-gff42f78b",
      "version.current.code": 36,
      "preference.update.lastTimestamp": 1765296218246,
      "version.previous.code": 35,
      "version.previous.name": "0.3.11-1-g0187646a"
    }
  },
  "PRODUCT": "mdarcy",
  "APP_VERSION_NAME": "0.3.12-1-gff42f78b",
  "USER_CRASH_DATE": "2026-01-20T15:20:53.482+01:00",
  "APP_VERSION_CODE": 36,
  "USER_COMMENT": "",
  "USER_EMAIL": ""
}
```

--- a/app/src/main/java/com/github/damontecres/wholphin/MainActivity.kt
+++ b/app/src/main/java/com/github/damontecres/wholphin/MainActivity.kt
@@ -150,7 +150,6 @@ class MainActivity : AppCompatActivity() {
             }
         }
         viewModel.appStart()
-        val requestedDestination = this.intent?.let(::extractDestination)
         setContent {
             val appPreferences by userPreferencesDataStore.data.collectAsState(null)
             appPreferences?.let { appPreferences ->
@@ -257,6 +256,10 @@ class MainActivity : AppCompatActivity() {
                                                     }
 
                                                     if (showContent) {
+                                                        val requestedDestination =
+                                                            remember(intent) {
+                                                                intent?.let(::extractDestination)
+                                                            }
                                                         ApplicationContent(
                                                             user = current.user,
                                                             server = current.server,
@@ -344,6 +347,7 @@ class MainActivity : AppCompatActivity() {
     override fun onNewIntent(intent: Intent) {
         super.onNewIntent(intent)
         Timber.v("onNewIntent")
+        setIntent(intent)
         extractDestination(intent)?.let {
             navigationManager.replace(it)
         }

--- a/app/src/main/java/com/github/damontecres/wholphin/services/NavigationManager.kt
+++ b/app/src/main/java/com/github/damontecres/wholphin/services/NavigationManager.kt
@@ -75,7 +75,11 @@ class NavigationManager
             while (backStack.size > 1) {
                 backStack.removeLastOrNull()
             }
-            backStack[0] = destination
+            if (backStack.isEmpty()) {
+                backStack.add(0, destination)
+            } else {
+                backStack[0] = destination
+            }
             log()
         }
 

